providers/logrus/provider_test.go

```diff
@@ -0,0 +1,45 @@
+// © 2016-2017 Helix OpCo LLC. All rights reserved.
+// Initial Author: Chris Williams
+
+package logrus
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/myhelix/contextlogger/log"
+	"github.com/myhelix/contextlogger/providers"
+	. "github.com/onsi/gomega"
+)
+
+var output *bytes.Buffer
+var testProvider providers.LogProvider
+
+func setupJSON(t *testing.T) {
+	RegisterTestingT(t)
+
+	output = new(bytes.Buffer)
+	provider, err := LogProvider(nil, Config{
+		output,
+		"debug",
+		&logrus.JSONFormatter{
+			DisableTimestamp: true,
+			TimestampFormat:  "sometime", // Omit timestamp to make output predictable
+		},
+	})
+	Expect(err).To(BeNil())
+	testProvider = provider
+}
+
+func TestJSONLogs(t *testing.T) {
+	setupJSON(t)
+
+	log.SetDefaultProvider(testProvider)
+
+	log.WithFields(log.Fields{
+		"first_name": "Sam",
+		"last_name":  "Gamgee",
+	}).Info("Hi there.")
+	Expect(output.String()).To(MatchRegexp(`{"first_name":"Sam","last_name":"Gamgee","level":"info","msg":"Hi there."}`))
```

Comment:
Would it be worth adding a test for multi-line/error output, if only to document clear expectations?